File: WebAPI/Models/NewOrderDto.cs
namespace WebAPI.Models
{
    public class NewOrderDto
    {
        public int EmpId { get; set; }
        public int CustId { get; set; }
        public int ShipperId { get; set; }
        public string ShipName { get; set; }
        public string ShipAddress { get; set; }
        public string ShipCity { get; set; }
        public DateTime RequiredDate { get; set; }
        public DateTime? ShippedDate { get; set; }
        public decimal Freight { get; set; }
        public string ShipCountry { get; set; }

        // detalle de la orden
        public NewOrderDetailDto OrderDetails { get; set; }
    }
}

File: WebAPP/src/app/models/newOrderDto.ts
import { NewOrderDetailDto } from "./newOrderDetailDto";

export class NewOrderDto {
  constructor(
    public empId: number,
    public custId: number,
    public shipperId: number,
    public shipName: string,
    public shipAddress: string,
    public shipCity: string,
    public orderdate: Date,
    public requiredDate: Date,
    public shippedDate: Date,
    public freight: number,
    public shipCountry: string,
    public orderDetails: NewOrderDetailDto
  ) {}
}
